docs/reference/reranking/index.asciidoc

```diff
@@ -0,0 +1,71 @@
+[[re-ranking-overview]]
+= Re-ranking
+
+Many search systems are built on two-stage retrieval pipelines.
+
+The first stage uses cheap, fast algorithms to find a subset of possible matches.
+
+The second stage uses a more powerful model, often machine learning-based, to reorder the documents.
+This second step is called re-ranking.
+This approach balances computational costs, because the resource-intensive model is only applied to a smaller set of pre-filtered results.
+
+{es} supports various ranking and re-ranking techniques to optimize search relevance and performance.
+
+[float]
+[[re-ranking-two-stage-pipeline]]
+== Two-stage retrieval pipelines
+
+Learn about retrieval pipelines and how re-ranking can be applied to your existing search experience.
+
+[float]
+[[re-ranking-first-stage-pipeline]]
+=== First stage: initial retrieval
+
+[float]
+[[re-ranking-ranking-overview-bm25]]
+==== Full-text search: BM25 scoring
+
+{es} ranks documents based on term frequency and inverse document frequency, adjusted for document length.
+BM25 is the default statistical scoring algorithm in {es} and works out-of-the-box.
+
+[float]
+[[re-ranking-ranking-overview-vector]]
+==== Vector search: similarity scoring
+
+Vector search involves transforming data into dense or sparse vector embeddings to capture semantic meanings, and computing similarity scores for query vectors.
+Store vectors using `semantic` fields for automatic vectorization or `dense_vector` and `sparse_vector` fields when you need more control over the underlying embedding model.
+Query vectors with `semantic` queries or `knn` and `sparse_vector` queries to compute similarity scores.
```

Comment:
```suggestion
Query vector fields with `semantic`, `knn` or `sparse_vector` queries to compute similarity scores.
```

Links to the query DSL pages for each query type would be helpful here. Also, I think we want to start specifying "retriever first", so maybe we should mention that here before we mention the queries?